src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -128,7 +132,7 @@ private static JsonConverter[] GetDefaultFactoryConverters()
             Add(JsonMetadataServices.UriConverter);
             Add(JsonMetadataServices.VersionConverter);
 
-            Debug.Assert(NumberOfSimpleConverters == converters.Count);
+            Debug.Assert(converters.Count <= NumberOfSimpleConverters);
```

Comment:
Consider keeping this assertion precise and assign `NumberOfSimpleConverters` according to `NET6_0_OR_GREATER` or not.

Reply:
I considered doing this, but it makes things needlessly complicated (needing to keep a separate tally of converters for each target) without any real benefits. Since we really only want to avoid resizing the dictionary on initialization, just keeping track of the upper bound should suffice.